web/static/js/reducers/user_vote_counter.js

```diff
@@ -0,0 +1,22 @@
+const userVoteCounter = (state = {}, action) => {
+  switch (action.type) {
+    case "SET_INITIAL_STATE":
+      return action.initialState.participations.reduce(
```

Comment:
hmmm... curious about the decision to introduce an object to house everyone's vote count. feels like YAGNI in that, at least for now, we only need the current user's vote count.

Reply:
so true, forgot that the `currentUser` is derived. I think adding a `voteCount` to each user on the store is the most sensible approach, as we'll get `currentUser.voteCount` for free *due* to the aforementioned derivation.

Reply:
Regarding a database refactor, no need for that now. Wasn't proposing that we change it now. I just want us to be vigilant about the state leaking into different corners of the app. if anyone notices it, they should raise the red flag.